core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushMergeWriterDeleteIntoConnector.java

```diff
@@ -64,7 +65,8 @@ public Result apply(TableFinishNode node, Captures captures, Context context)
                 .map(newHandle -> new TableDeleteNode(
                         context.getIdAllocator().getNextId(),
                         newHandle,
-                        getOnlyElement(node.getOutputSymbols())))
+                        getOnlyElement(node.getOutputSymbols()),
+                        ImmutableList.of(tableScan.getTable())))
```

Comment:
Isn't the `tableScan.getTable() target table?

Reply:
The scanned table is what is being read.
See also a few lines above which indicates as well that this is the input table.

```
metadata.applyDelete(context.getSession(), tableScan.getTable())
```